Library/Formula/mandoc.rb

```diff
@@ -0,0 +1,81 @@
+class Mandoc < Formula
+  homepage "http://mdocml.bsd.lv"
+  url "http://mdocml.bsd.lv/snapshots/mdocml-1.13.3.tar.gz"
+  sha256 "23ccab4800d50bf4c327979af5d4aa1a6a2dc490789cb67c4c3ac1bd40b8cad8"
+
+  head "anoncvs@mdocml.bsd.lv:/cvs", :module => "mdocml", :using => :cvs
+
+  depends_on "sqlite"
+
+  option "without-db", "Only install the mandoc/demandoc utilities."
+  option "without-cgi", "Don't build man.cgi (and extra CSS files)."
+
+  def install
+    localconfig = [
+
+      # Sane prefixes.
+      "PREFIX=#{prefix}",
+      "INCLUDEDIR=#{include}",
+      "LIBDIR=#{lib}",
+      "MANDIR=#{man}",
+      "WWWPREFIX=#{prefix}/var/www",
+      "EXAMPLEDIR=#{share}/examples",
+
+      # Executable names, where utilites would be replaced/dupelicated.
+      # The mdocml versions of the utilities are definitely *not* ready
+      # for prime-time on Darwin, though some changes in HEAD are promising.
+      # The "bsd" prefix (like bsdtar, bsdmake) is more informative than "m".
+      "BINM_MAN=bsdman",
+      "BINM_APROPOS=bsdapropos",
+      "BINM_WHATIS=bsdwhatis",
+      "BINM_MAKEWHATIS=bsdmakewhatis",	# default is "makewhatis".
+
+      # These are names for *section 7* pages only. Several other pages are
+      # prefixed "mandoc_", similar to the "groff_" pages.
+      "MANM_MAN=man",
+      "MANM_MDOC=mdoc",
+      "MANM_ROFF=mandoc_roff", # This is the only one that conflicts (groff).
+      "MANM_EQN=eqn",
+      "MANM_TBL=tbl",
+
+      "OSNAME='Mac OS X #{MacOS.version}'", # Bottom corner signature line.
+
+      # Not quite sure what to do here. The default ("/usr/share", etc.) needs
+      # sudoer privileges, or will error. So just brew's manpages for now?
+      "MANPATH_DEFAULT=#{HOMEBREW_PREFIX}/share/man",
+
+      "HAVE_MANPATH=0",   # Our `manpath` is a symlink to system `man`.
+      "STATIC=",          # No static linking on Darwin.
+
+      "HOMEBREWDIR=#{HOMEBREW_CELLAR}" # ?? See configure.local.example, NEWS.
+    ]
+
+    localconfig << "BUILD_DB=1" if build.with? "db"
+    localconfig << "BUILD_CGI=1" if build.with? "cgi"
+    File.rename("cgi.h.example", "cgi.h") # For man.cgi, harmless in any case.
+
+    (buildpath/"configure.local").write localconfig.join("\n")
+    system "./configure"
+
+    ENV.deparallelize
+
+    system "make"
+    system "make", "install"
+
+    system "make", "manpage" # Left out of the install for some reason.
+    bin.install "manpage"
+  end
+
+  test do
+    system "mandoc", "-Thtml",
+    "-Ostyle=#{share}/examples/example.style.css",
+    "#{HOMEBREW_PREFIX}/share/man/man1/brew.1"
+
+    # I know Mike's probably gonna make me get rid of these, but... ;)
+    (testpath/"brew.1.html").write(`mandoc -Thtml  -Ostyle=#{opt_prefix}/var/www/htdocs/man.css #{HOMEBREW_PREFIX}/share/man/man1/brew.1`)
+    (testpath/"brew.1.pdf").write(`mandoc -Tpdf -Opaper=a4 #{HOMEBREW_PREFIX}/share/man/man1/brew.1`)
+    system "open", "brew.1.html"
+    system "open", "brew.1.pdf"
```

Comment:
Well, you beat Mike to it  :wink:.

My thought here is that without the loginwindow process running (as on the testbot, I assume) `open` will _not_ spawn a new process; and when run from an interactive terminal, assumedly the user will close Preview/the browser. No? Is there way you can think of to do a non-CLI visual test? Or should I just dump the HTML and call it a day?